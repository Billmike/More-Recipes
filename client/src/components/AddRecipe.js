import React from 'react';
import RecipesForm from './RecipesForm';
import { connect } from 'react-redux';
import { addRecipe } from '../actions/recipes';

const AddRecipe = (props) => {
    console.log(props)
	return (
	   <div>
       <h1 className="container">Add Recipe</h1>
        <RecipesForm onSubmit={(recipe) => {
            props.dispatch(addRecipe(recipe));
            props.history.push('/')
        }} />
       </div>	
	)
}

export default connect()(AddRecipe);
