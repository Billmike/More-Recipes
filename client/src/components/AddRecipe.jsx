import React, { Component } from 'react';
import RecipesForm from './RecipesForm';
import { connect } from 'react-redux';
import { addRecipe } from '../actions/recipes';

export class AddRecipe extends Component {
  onSubmit = (recipe) => {
    this.props.addRecipe(recipe);
    this.props.history.push('/dashboard');
  };

  render() {
    return (
      <div>
        <h1 className="container"> Add Recipe </h1>
        <RecipesForm
          onSubmit={this.onSubmit}
        />
      </div>
      );
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    addRecipe: (recipe) => dispatch(addRecipe(recipe))
  }
}

// const AddRecipe = (props) => {
//     console.log(props)
// 	return (
// 	   <div>
//        <h1 className="container">Add Recipe</h1>
//         <RecipesForm onSubmit={(recipe) => {
//             props.dispatch(addRecipe(recipe));
//             props.history.push('/')
//         }} />
//        </div>	
// 	)
// }

export default connect(undefined, mapDispatchToProps)(AddRecipe);
