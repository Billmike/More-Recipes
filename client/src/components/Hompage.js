import React, { Component } from 'react';
import '../assets/fonts/css/font-awesome.min.css';
import pancakes from '../assets/img/pancakes.jpeg';
import cookies from '../assets/img/cookies.jpeg';
import dessert from '../assets/img/dessert.jpeg';
import pizza from '../assets/img/pizzza.jpg';

const Homepage = () => (
  <div>
    <div id="carouselExampleIndicators" className="carousel slide" data-ride="carousel">
            <ol className="carousel-indicators">
          <li data-target="#carouselExampleIndicators" data-slide-to="0" className="active" />
          <li data-target="#carouselExampleIndicators" data-slide-to="1" />
          <li data-target="#carouselExampleIndicators" data-slide-to="2" />
        </ol>
            <div className="carousel-inner">
          <div className="carousel-item active">
              <img className="d-block w-100 sizing" src={pancakes} alt="First slide" />
              <div className="carousel-caption d-none d-md-block">
                <h3 className="carousel-h3">Handcrafted recipes made by local chefs</h3>
                <p className="carousel-p">... the little things matter</p>
              </div>
            </div>
          <div className="carousel-item">
              <img className="d-block w-100 sizing" src={cookies} alt="Second slide" />
              <div className="carousel-caption d-none d-md-block">
                <h3 className="carousel-h3-light">Tasty recipes to spice up your dinner table.</h3>
                <p className="carousel-p-light">... you really can't resist these delights.</p>
              </div>
            </div>
          <div className="carousel-item">
              <img className="d-block w-100 sizing" src={dessert} alt="Third slide" />
              <div className="carousel-caption d-none d-md-block">
                <h3 className="carousel-h3">Make every meal memorable.</h3>
                <p className="carousel-p">... no more boring nights at home.</p>
              </div>
            </div>
        </div>
            <a className="carousel-control-prev" href="#carouselExampleIndicators" role="button" data-slide="prev">
          <span className="carousel-control-prev-icon" aria-hidden="true" />
          <span className="sr-only">Previous</span>
        </a>
            <a className="carousel-control-next" href="#carouselExampleIndicators" role="button" data-slide="next">
          <span className="carousel-control-next-icon" aria-hidden="true" />
          <span className="sr-only">Next</span>
        </a>
          </div>

    <div className="card-group">
        <div className="card index-card">
            <a href="./template/recipe_detail.html"><img className="card-img-top" src={ pizza } alt="Card image cap" /></a>
            <div className="card-body">
              <a href="./template/recipe_detail.html"><h4 className="card-title">Pizza Pepperoni with mushroom toppings.</h4></a>
              <p className="card-text">This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.</p>
              <a href="./template/recipe_detail.html" className="btn border border-secondary rounded">45 <i className="fa fa-eye" aria-hidden="true" /></a>
              <a href="#" className="btn border border-secondary rounded" data-toggle="modal" data-target="#exampleModal">  89 <i className="fa fa-heart" aria-hidden="true" /></a>
              <a href="./template/recipe_detail.html#comment" className="btn border border-secondary rounded"> 6 <i className="fa fa-comments" aria-hidden="true" /></a>
            </div>
            <div className="card-footer">
              <small className="text-muted">Last updated 3 mins ago</small>
            </div>
          </div>
        <div className="card index-card">
            <a href="./template/recipe_detail.html"><img className="card-img-top" src={ pizza } alt="Card image cap" /></a>
            <div className="card-body">
              <a href="./template/recipe_detail.html"><h4 className="card-title">Pizza Pepperoni with mushroom toppings.</h4></a>
              <p className="card-text">This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.</p>
              <a href="./template/recipe_detail.html" className="btn border border-secondary rounded">180 <i className="fa fa-eye" aria-hidden="true" /></a>
              <a href="#" className="border border-secondary rounded btn" data-toggle="modal" data-target="#exampleModal">  150 <i className="fa fa-heart" aria-hidden="true" /></a>
              <a href="./template/recipe_detail.html#comment" className="btn border border-secondary rounded"> 70 <i className="fa fa-comments" aria-hidden="true" /></a>
            </div>
            <div className="card-footer">
              <small className="text-muted">Last updated 3 mins ago</small>
            </div>
          </div>
        <div className="card index-card">
            <a href="./template/recipe_detail.html"><img className="card-img-top" src={ pizza } alt="Card image cap" /></a>
            <div className="card-body">
              <a href="./template/recipe_detail.html"><h4 className="card-title">Pizza Pepperoni with mushroom toppings.</h4></a>
              <p className="card-text">This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action. </p>
              <a href="./template/recipe_detail.html" className="border btn border-secondary rounded">67 <i className="fa fa-eye" aria-hidden="true" /></a>
              <a href="#" className="border btn border-secondary rounded" data-toggle="modal" data-target="#exampleModal">  49 <i className="fa fa-heart" aria-hidden="true" /></a>
              <a href="./template/recipe_detail.html#comment" className="btn border border-secondary rounded"> 90 <i className="fa fa-comments" aria-hidden="true" /></a>
            </div>
            <div className="card-footer">
              <small className="text-muted">Last updated 3 mins ago</small>
            </div>
          </div>
      </div>
  </div>
);

export default Homepage;
