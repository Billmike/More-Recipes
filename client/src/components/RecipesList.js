import React from 'react';
import pizza from '../assets/img/pizzza.jpg';

const RecipeList = ({ name, description, category }) => {
	return (
		<div className="card-group">
        <div className="card index-card">
            <a href="./template/recipe_detail.html"><img className="card-img-top" src={ pizza } alt="Card image cap" /></a>
            <div className="card-body">
              <a href="./template/recipe_detail.html"><h4 className="card-title">{name}</h4></a>
              <p className="card-text">{description}</p>
              <a href="./template/recipe_detail.html" className="btn border border-secondary rounded">45 <i className="fa fa-eye" aria-hidden="true" /></a>
              <a href="#" className="btn border border-secondary rounded" data-toggle="modal" data-target="#exampleModal">  89 <i className="fa fa-heart" aria-hidden="true" /></a>
              <a href="./template/recipe_detail.html#comment" className="btn border border-secondary rounded"> 6 <i className="fa fa-comments" aria-hidden="true" /></a>
            </div>
            <div className="card-footer">
              <small className="text-muted">Recipe Category - {category}</small>
            </div>
          </div>
          <div className="card index-card">
            <a href="./template/recipe_detail.html"><img className="card-img-top" src={ pizza } alt="Card image cap" /></a>
            <div className="card-body">
              <a href="./template/recipe_detail.html"><h4 className="card-title">{name}</h4></a>
              <p className="card-text">{description}</p>
              <a href="./template/recipe_detail.html" className="btn border border-secondary rounded">45 <i className="fa fa-eye" aria-hidden="true" /></a>
              <a href="#" className="btn border border-secondary rounded" data-toggle="modal" data-target="#exampleModal">  89 <i className="fa fa-heart" aria-hidden="true" /></a>
              <a href="./template/recipe_detail.html#comment" className="btn border border-secondary rounded"> 6 <i className="fa fa-comments" aria-hidden="true" /></a>
            </div>
            <div className="card-footer">
              <small className="text-muted">Recipe Category - {category}</small>
            </div>
          </div>
          <div className="card index-card">
            <a href="./template/recipe_detail.html"><img className="card-img-top" src={ pizza } alt="Card image cap" /></a>
            <div className="card-body">
              <a href="./template/recipe_detail.html"><h4 className="card-title">{name}</h4></a>
              <p className="card-text">{description}</p>
              <a href="./template/recipe_detail.html" className="btn border border-secondary rounded">45 <i className="fa fa-eye" aria-hidden="true" /></a>
              <a href="#" className="btn border border-secondary rounded" data-toggle="modal" data-target="#exampleModal">  89 <i className="fa fa-heart" aria-hidden="true" /></a>
              <a href="./template/recipe_detail.html#comment" className="btn border border-secondary rounded"> 6 <i className="fa fa-comments" aria-hidden="true" /></a>
            </div>
            <div className="card-footer">
              <small className="text-muted">Recipe Category - {category}</small>
            </div>
          </div>
      </div>
		)
};

export default RecipeList;
