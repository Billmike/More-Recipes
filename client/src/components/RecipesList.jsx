import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

export const RecipeList = ({ id, name, description, category }) => (
  <div className="card-group">
      <div className="card index-card">
        <Link to={`/recipe/${id}`}><img className="card-img-top" alt="Pizza" /></Link>
        <div className="card-body">
          <Link to={`/recipe/${id}`}><h4 className="card-title">{name}</h4></Link>
          <p className="card-text">{description}</p>
          <Link to={`/recipe/${id}`} className="btn border border-secondary rounded">45 <i className="fa fa-eye" aria-hidden="true" /></Link>
          <Link to={`/recipe/${id}`} className="btn border border-secondary rounded">  89 <i className="fa fa-heart" aria-hidden="true" /></Link>
          <Link to={`/recipe/${id}`} className="btn border border-secondary rounded"> 6 <i className="fa fa-comments" aria-hidden="true" /></Link>
        </div>
        <div className="card-footer">
          <small className="text-muted">Recipe Category - {category}</small>
        </div>
      </div>
      <div className="card index-card">
        <Link to={`/recipe/${id}`}><img className="card-img-top" alt="Pizza" /></Link>
        <div className="card-body">
          <Link to={`/recipe/${id}`}><h4 className="card-title">{name}</h4></Link>
          <p className="card-text">{description}</p>
          <Link to={`/recipe/${id}`} className="btn border border-secondary rounded">45 <i className="fa fa-eye" aria-hidden="true" /></Link>
          <Link to={`/recipe/${id}`} className="btn border border-secondary rounded">  89 <i className="fa fa-heart" aria-hidden="true" /></Link>
          <Link to={`/recipe/${id}`} className="btn border border-secondary rounded"> 6 <i className="fa fa-comments" aria-hidden="true" /></Link>
        </div>
        <div className="card-footer">
          <small className="text-muted">Recipe Category - {category}</small>
        </div>
      </div>
      <div className="card index-card">
        <Link to={`/recipe/${id}`}><img className="card-img-top" alt="Pizza" /></Link>
        <div className="card-body">
          <Link to={`/recipe/${id}`}><h4 className="card-title">{name}</h4></Link>
          <p className="card-text">{description}</p>
          <Link to={`/recipe/${id}`} className="btn border border-secondary rounded">45 <i className="fa fa-eye" aria-hidden="true" /></Link>
          <Link to={`/recipe/${id}`} className="btn border border-secondary rounded">  89 <i className="fa fa-heart" aria-hidden="true" /></Link>
          <Link to={`/recipe/${id}`} className="btn border border-secondary rounded"> 6 <i className="fa fa-comments" aria-hidden="true" /></Link>
        </div>
        <div className="card-footer">
          <small className="text-muted">Recipe Category - {category}</small>
        </div>
      </div>
  </div>
);

export default connect()(RecipeList);
