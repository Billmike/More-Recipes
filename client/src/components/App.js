import React, { Component } from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import Navbar from './Navbar';
import Homepage from './Hompage';
import Dashboard from './Dashboard';
import Favorite from './Favorite';
import AddRecipe from './AddRecipe';
import EditRecipe from './EditRecipe';
import ProfilePage from './ProfilePage';
import NotFoundPage from './NotFoundPage';

class App extends Component {
    render() {
        return(
            <div>
                <BrowserRouter>
                    <div>
                        <Navbar />
                        <Switch>
                        <Route exact path="/" component={Homepage} />
                        <Route path="/dashboard" component={Dashboard} />
                        <Route path="/favorite" component={Favorite} />
                        <Route path="/add" component={AddRecipe} />
                        <Route path="/edit" component={EditRecipe} />
                        <Route path="/profile" component={ProfilePage} />
                        <Route component={NotFoundPage} />
                    </Switch>
                    </div>
                </BrowserRouter>
            </div>
        )
    }
}

export default App;
