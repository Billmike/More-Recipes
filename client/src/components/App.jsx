import React from 'react';
import { BrowserRouter, browserHistory, Route, Switch } from 'react-router-dom';
import Navbar from './Navbar';
import Homepage from './Hompage';
import Dashboard from './Dashboard';
import RecipeDetails from './RecipeDetails';
import Favorite from './Favorites';
import AddRecipe from './AddRecipe';
import EditRecipe from './EditRecipe';
import ProfilePage from './ProfilePage';
import LoginPage from './LoginPage';
import RegistrationPage from './RegistrationPage';
import RequireAuth from '../utils/RequireAuth';
import NotFoundPage from './NotFoundPage';

const App = () => (
  <div>
    <BrowserRouter history={browserHistory}>
      <div>
        <Navbar />
        <Switch>
          <Route exact path="/" component={Homepage} />
          <Route path="/dashboard" component={RequireAuth(Dashboard)} />
          <Route path="/favorites" component={RequireAuth(Favorite)} />
          <Route path="/add" component={RequireAuth(AddRecipe)} />
          <Route path="/edit/:id" component={RequireAuth(EditRecipe)} />
          <Route path="/recipe/:id" component={RecipeDetails} />
          <Route path="/profile" component={RequireAuth(ProfilePage)} />
          <Route path="/login" component={LoginPage} />
          <Route path="/register" component={RegistrationPage} />
          <Route component={NotFoundPage} />
        </Switch>
      </div>
    </BrowserRouter>
  </div>
);

export default App;
