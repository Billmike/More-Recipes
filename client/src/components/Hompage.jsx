import React, { Component } from 'react';
import { connect } from 'react-redux';
import Emoji from 'react-emoji-render';
import ReactPaginate from 'react-paginate';
import '../assets/css/style.css';
import LoaderComp from './LoaderComp';
import Loader from './Loader';
import RecipesList from './RecipesList';
import Footer from './Footer';
import Pagination from './Pagination';
import {
  GetAllRecipesAction,
  GetPopularRecipes, SearchRecipesAction
} from '../actions/recipes';
import strawberry from '../assets/img/strawberry.jpg';
import dark from '../assets/img/dark.jpg';
import noodles from '../assets/img/noodles.jpg';


/**
 * Component that renders the Homepage
 *
 * @class Homepage
 *
 * @extends Component
 */

export class Homepage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      searchQuery: '',
      loaded: true,
      page: 1,
      search: false
    };
    this.onSearch = this.onSearch.bind(this);
    this.handlePaginationChange = this.handlePaginationChange.bind(this);
  }
  componentDidMount() {
    this.props.GetAllRecipesAction(this.state.page);
    this.props.GetPopularRecipes();
  }

  onSearch(event) {
    this.setState({ [event.target.name]: event.target.value, search: true });
    this.props.SearchRecipesAction(event.target.value, this.state.page);
  }

  handlePaginationChange(data) {
    const currentView = data.selected;
    this.setState({ page: currentView });
    if (this.state.search) {
      this.props.SearchRecipesAction(this.state.searchQuery, currentView);
    } else {
      this.props.GetAllRecipesAction(currentView);
    }
  }

  render() {
    let allRecipes;
    let popularRecipesList;
    if (this.props.isLoading) {
      return (
        <Loader />
      );
    }
    if (this.props.recipes) {
      allRecipes = this.props.recipes.length > 0 ?
        this.props.recipes.map((recipe) => {
          return <RecipesList key={recipe.id} recipe={recipe} />;
        }) : (
          <div>
            <p className="search-text">
              No results found for your search <Emoji text=";(" />
            </p>
          </div>
        );
    }
    if (this.props.popularRecipes) {
      popularRecipesList = this.props.popularRecipes.map((recipe) => {
        return <RecipesList key={recipe.id} recipe={recipe} />;
      });
    }
    return (
      <div className="under-carousel-div">
        <div
          id="carouselExampleIndicators"
          className="carousel slide"
          data-ride="carousel"
        >
          <ol className="carousel-indicators">
            <li
              data-target="#carouselExampleIndicators"
              data-slide-to="0"
              className="active"
            />
            <li data-target="#carouselExampleIndicators" data-slide-to="1" />
            <li data-target="#carouselExampleIndicators" data-slide-to="2" />
          </ol>
          <div className="carousel-inner">
            <div className="carousel-item active">
              <img
                className="d-block w-100 sizing carousel-slide-image"
                src={strawberry}
                alt="First slide"
              />
              <div className="carousel-caption d-none d-md-block">
                <h3 className="carousel-h3">
                  Handcrafted recipes made by local chefs
                </h3>
              </div>
            </div>
            <div className="carousel-item">
              <img
                className="d-block w-100 sizing carousel-slide-image"
                src={dark}
                alt="Second slide"
              />
              <div className="carousel-caption d-none d-md-block">
                <h3 className="carousel-h3">
                  Tasty recipes to spice up your dinner table.
                </h3>
              </div>
            </div>
            <div className="carousel-item">
              <img
                className="d-block w-100 sizing carousel-slide-image"
                src={noodles}
                alt="Third slide"
              />
              <div className="carousel-caption d-none d-md-block">
                <h3 className="carousel-h3">Make every meal memorable.</h3>
              </div>
            </div>
          </div>
          <a
            className="carousel-control-prev"
            href="#carouselExampleIndicators"
            role="button"
            data-slide="prev"
          >
            <span className="carousel-control-prev-icon" aria-hidden="true" />
            <span className="sr-only">Previous</span>
          </a>
          <a
            className="carousel-control-next"
            href="#carouselExampleIndicators"
            role="button"
            data-slide="next"
          >
            <span className="carousel-control-next-icon" aria-hidden="true" />
            <span className="sr-only">Next</span>
          </a>
        </div>
        <div className="container">
          <h2 className="homepage-h2"> Popular Recipes of the week</h2>
          <div className="row">{popularRecipesList}</div>
          <hr className="frontpage-hr" />
          <div className="input-group search-button">
            <span className="input-group-btn">

            </span>
            <input
              type="search"
              name="searchQuery"
              id="searchParamId"
              value={this.state.searchQuery}
              className="form-control"
              placeholder="Search for....."
              onChange={this.onSearch}
            />
          </div>
          <h2 className="homepage-h2"> Recipe Catalogue</h2>
          <div className="row">{allRecipes}</div>
          <Pagination
            handlePaginationChange={this.handlePaginationChange}
            pageCount={this.props.pages}
          />
        </div>
        <Footer />
      </div>
    );
  }
}

export const mapStateToProps = (state) => {
  console.log('current state', state);
  return {
    popularRecipes: state.recipes.popularRecipes,
    recipes: state.recipes.recipes,
    pages: state.recipes.pages,
    isLoading: state.recipes.isLoading
  };
};

export default connect(
  mapStateToProps,
  {
    GetAllRecipesAction, GetPopularRecipes, SearchRecipesAction
  }
)(Homepage);
