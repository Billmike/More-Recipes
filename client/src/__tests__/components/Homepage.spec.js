import React from 'react';
import { shallow } from 'enzyme';
import { Homepage, mapStateToProps } from '../../components/Hompage';
import { RecipeList } from '../../components/RecipesList';
import state from '../fixtures/state';

describe('<Homepage />', () => {
  it('Should render the homepage component correctly', () => {
    const GetAllRecipesAction = jest.fn();
    const GetPopularRecipes = jest.fn();
    const SearchRecipesAction = jest.fn();
    const wrapper = shallow(<Homepage
      GetAllRecipesAction={() => Promise.resolve()}
      GetPopularRecipes={() => Promise.resolve()}
      SearchRecipesAction={() => Promise.resolve()}
    />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').length).toBe(13);
  });

  it('Should render RecipesList as a child element', () => {
    const wrapper = shallow(<Homepage
      GetAllRecipesAction={() => Promise.resolve()}
      GetPopularRecipes={() => Promise.resolve()}
      SearchRecipesAction={() => Promise.resolve()}
    >
      <div>
        <h2 className="homepage-h2"> Popular Recipes of the week</h2>
        <div className="row">
          <RecipeList />
        </div>
      </div>
    </Homepage>);
  });

  it('Should call mapStateToProps', (done) => {
    mapStateToProps(state);
    done();
  });
});
