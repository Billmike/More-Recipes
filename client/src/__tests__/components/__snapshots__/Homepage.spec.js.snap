// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Homepage /> Should render the homepage component correctly 1`] = `
<div
  className="under-carousel-div"
>
  <div
    className="carousel slide"
    data-ride="carousel"
    id="carouselExampleIndicators"
  >
    <ol
      className="carousel-indicators"
    >
      <li
        className="active"
        data-slide-to="0"
        data-target="#carouselExampleIndicators"
      />
      <li
        data-slide-to="1"
        data-target="#carouselExampleIndicators"
      />
      <li
        data-slide-to="2"
        data-target="#carouselExampleIndicators"
      />
    </ol>
    <div
      className="carousel-inner"
    >
      <div
        className="carousel-item active"
      >
        <img
          alt="First slide"
          className="d-block w-100 sizing carousel-slide-image"
          src="test-file-stub"
        />
        <div
          className="carousel-caption d-none d-md-block"
        >
          <h3
            className="carousel-h3"
          >
            Handcrafted recipes made by local chefs
          </h3>
        </div>
      </div>
      <div
        className="carousel-item"
      >
        <img
          alt="Second slide"
          className="d-block w-100 sizing carousel-slide-image"
          src="test-file-stub"
        />
        <div
          className="carousel-caption d-none d-md-block"
        >
          <h3
            className="carousel-h3"
          >
            Tasty recipes to spice up your dinner table.
          </h3>
        </div>
      </div>
      <div
        className="carousel-item"
      >
        <img
          alt="Third slide"
          className="d-block w-100 sizing carousel-slide-image"
          src="test-file-stub"
        />
        <div
          className="carousel-caption d-none d-md-block"
        >
          <h3
            className="carousel-h3"
          >
            Make every meal memorable.
          </h3>
        </div>
      </div>
    </div>
    <a
      className="carousel-control-prev"
      data-slide="prev"
      href="#carouselExampleIndicators"
      role="button"
    >
      <span
        aria-hidden="true"
        className="carousel-control-prev-icon"
      />
      <span
        className="sr-only"
      >
        Previous
      </span>
    </a>
    <a
      className="carousel-control-next"
      data-slide="next"
      href="#carouselExampleIndicators"
      role="button"
    >
      <span
        aria-hidden="true"
        className="carousel-control-next-icon"
      />
      <span
        className="sr-only"
      >
        Next
      </span>
    </a>
  </div>
  <div
    className="container"
  >
    <div
      className="input-group search-button"
    >
      <span
        className="input-group-btn"
      />
      <input
        className="form-control"
        name="searchQuery"
        onChange={[Function]}
        placeholder="Search for....."
        type="search"
        value=""
      />
    </div>
    <h2
      className="homepage-h2"
    >
       Popular Recipes of the week
    </h2>
    <div
      className="row"
    />
    <hr
      className="frontpage-hr"
    />
    <h2
      className="homepage-h2"
    >
       Recipe Catalogue
    </h2>
    <div
      className="row"
    />
    <Pagination
      handlePaginationChange={[Function]}
    />
  </div>
  <Footer />
</div>
`;
